processor/sourcemap/processor.go

```diff
@@ -33,36 +35,31 @@ func NewProcessor() pr.Processor {
 	return &processor{schema}
 }
 
-func (p *processor) Validate(buf []byte) error {
+func (p *processor) Validate(raw map[string]interface{}) error {
 	validationCount.Inc()
 
-	var pa payload
-	err := json.Unmarshal(buf, &pa)
+	bytes, err := json.Marshal(raw["sourcemap"])
```

Comment:
in `DecodeSourcemapFormData` we decode the sourcemap to a map and here we decode it back. Would it be possible to just read the file to a `[]byte`, put the `[]byte` in the returned payload use it here in parser.Parse here?
Just an idea.

Reply:
Not easily. It would make testing quite more painful.